nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -765,16 +768,16 @@ def build_model_parallel_config(self) -> ModelParallelConfig:
         config_mapping = {
             "perform_initialization": True,  # initailize weights when constructing the module
             "fp16": False,  # NeMo does not currently support fp16 training with megatron amp O2
-            "bf16": precision == 'bf16' and megatron_amp_O2,
+            "bf16": self.torch_dtype == torch.bfloat16 and megatron_amp_O2,
```

Comment:
Just noting here that this is right for training but not necessary for inference to have megatron_amp_O2 flag.